cvat/apps/engine/views.py

```diff
@@ -2788,6 +2793,67 @@ class AnnotationGuidesViewSet(
     ordering = "-id"
     iam_organization_field = None
 
+    def _update_related_assets(self, request, guide: AnnotationGuide):
```

Comment:
There are many db objects being modified here, so I suggest wrapping this function as an atomic transaction

Reply:
I added transaction.atomic where we copy existing assets